my-app/src/app/core/services/auth.service.spec.ts

```diff
@@ -0,0 +1,50 @@
+import {TestBed} from '@angular/core/testing';
+
+import {AuthService} from './auth.service';
+
+import Spy = jasmine.Spy;
+
+describe('AuthServiceService', () => {
+  let service: AuthService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(AuthService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should call generateToken(), ' +
+    'assign true to IsAuthenticated variable, have email and token values in localStorage by login', () => {
+    const email: string = 'email@gmail.com';
+    const generationTokenSpy: Spy = spyOn(service, 'generateToken');
+
+    service.login(email);
+
+    expect(localStorage.getItem('email')).toEqual(email);
+    expect(generationTokenSpy).toHaveBeenCalled();
+    expect(service.IsAuthenticated).toBeTruthy();
+  });
+
+  it('should return string at least 20 characters long by calling generateToken()', () => {
+    const result = service.generateToken();
+
+    expect(typeof result).toBe('string');
+    expect(result.length).toBeGreaterThan(20);
+  });
+
+  it('should remove email and token fields from localStorage ' +
+    'and assign false to the IsAuthenticated by logout', () => {
+    const email: string = 'email@gmail.com';
+
+    service.IsAuthenticated = true;
+    service.login(email);
+    service.logout();
+
+    expect(localStorage.getItem('email')).toBeNull();
+    expect(localStorage.getItem('token')).toBeNull();
```

Comment:
Local storage is an external thing for the app, so it's not quite right to use it from the unit tests. Ideally, it should mocked/stubbed, and only calls to it should be tested (e.g. `expect(getItemSpy).toHaveBeenCalledWith(...)`). But it's a kind of purism, and for now current approach is ok.

Reply:
Added mock file for localStorage and changed testing of it